### PR TITLE
#37574 extend document search possibilities

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: document/src/main/java/com/ritense/document/domain/search/SearchRequestValue.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.ritense.valtimo.contract.utils.AssertionConcern.assertArgumentFalse;
+import static com.ritense.valtimo.contract.utils.AssertionConcern.assertArgumentTrue;
+
+/**
+ *
+ */
+public class SearchRequestValue {
```
Comment (Rick-Ritense) [inline document/src/main/java/com/ritense/document/domain/search/SearchRequestValue.java:32]: Empty comment?

File: document/src/test/resources/config/application.yml [hunk 2]
```diff
@@ -46,3 +46,8 @@ valtimo:
     versioning.enabled: true
     plugin:
         encryption-secret: "abcdefghijklmnop"
+
+logging:
+    level:
+        org.hibernate.SQL: DEBUG
+        org.hibernate.type: TRACE
```
Comment (Rick-Ritense) [inline document/src/test/resources/config/application.yml:53]: Do we want to keep this?
Seems obsessive for non-development purposes

Comment (Klaas-Ritense) [reply document/src/test/resources/config/application.yml:53]: This is used in the integration-tests and is very useful for debugging issues with the document search

File: contract/src/main/java/com/ritense/valtimo/contract/database/QueryDialectHelper.java [hunk 3]
```diff
@@ -23,7 +23,9 @@
 
 public interface QueryDialectHelper {
 
-    Expression<?> getJsonValueExpression(CriteriaBuilder cb, Path column, String path);
+    Expression<String> getJsonValueExpression(CriteriaBuilder cb, Path column, String path);
```
Comment (ivo-ritense) [inline contract/src/main/java/com/ritense/valtimo/contract/database/QueryDialectHelper.java:26]: Is there a reason why this is 2 different methods? Can't we just pass in the string class for the other one and get rid of this method?

File: document/src/main/java/com/ritense/document/domain/search/SearchRequest2.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+ */
+
+package com.ritense.document.domain.search;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SearchRequest2 {
+    private SearchOperator searchOperator = SearchOperator.OR;
```
Comment (ivo-ritense) [inline document/src/main/java/com/ritense/document/domain/search/SearchRequest2.java:24]: i would say the default should be AND

File: document/src/main/java/com/ritense/document/service/impl/JsonSchemaDocumentSearchService.java [hunk 5]
```diff
@@ -158,44 +244,238 @@ private void addJsonFieldPredicates(CriteriaBuilder cb, Root<JsonSchemaDocument>
...
+        } else {
+            throw new IllegalArgumentException("Search path doesn't start with known prefix: '" + searchCriteria.getPath() + "'");
+        }
+
+        var rangeFrom = searchCriteria.<T>getRangeFrom();
+        var rangeTo = searchCriteria.<T>getRangeTo();
+
+        switch (searchCriteria.getSearchType()) {
+            case LIKE:
+                return cb.or(searchLike(cb, value, searchCriteria.getValues()));
```
Comment (ivo-ritense) [inline document/src/main/java/com/ritense/document/service/impl/JsonSchemaDocumentSearchService.java:306]: Why is the `cb.or` here? Do that still work with an and search?

Comment (ivo-ritense) [reply document/src/main/java/com/ritense/document/service/impl/JsonSchemaDocumentSearchService.java:306]: I guess it's to support lists of strings?

Comment (Klaas-Ritense) [reply document/src/main/java/com/ritense/document/service/impl/JsonSchemaDocumentSearchService.java:306]: Yes, its to support this:
```
new SearchRequest2().addOtherFilters(new SearchRequest2.SearchCriteria2()
    .path("doc:customer.fullName")
    .searchType(LIKE)
    .addValue("John D")
    .addValue("Jane D"))
```

File: contract/src/main/java/com/ritense/valtimo/contract/database/QueryDialectHelper.java (hunk 3 above)
Comment (Klaas-Ritense) [reply contract/src/main/java/com/ritense/valtimo/contract/database/QueryDialectHelper.java:26]: Yes, the method is no longer needed. But there is one project that uses this method still. So removing it is a breaking change

File: document/src/main/java/com/ritense/document/domain/search/SearchRequest2.java (hunk 4 above)
Comment (Klaas-Ritense) [reply document/src/main/java/com/ritense/document/domain/search/SearchRequest2.java:24]: Changed

**Pull request merged**